controllers/managedfusiondeployment_controller.go

```diff
@@ -0,0 +1,752 @@
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.k8sMetricsServiceMonitor, func() error {
+		if err := r.own(r.k8sMetricsServiceMonitor); err != nil {
+			return err
+		}
+		desired := templates.K8sMetricsServiceMonitorTemplate.DeepCopy()
+		r.k8sMetricsServiceMonitor.Spec = desired.Spec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Failed to update k8sMetricsServiceMonitor: %v", err)
+	}
+	return nil
+}
+
+// reconcileMonitoringResources labels all monitoring resources (ServiceMonitors, PodMonitors, and PrometheusRules)
+// found in the target namespace with a label that matches the label selector the defined on the Prometheus resource
+// we are reconciling in reconcilePrometheus. Doing so instructs the Prometheus instance to notice and react to these labeled
+// monitoring resources
+func (r *ManagedFusionReconciler) reconcileMonitoringResources() error {
+	r.Log.Info("reconciling monitoring resources")
+
+	podMonitorList := promv1.PodMonitorList{}
+	if err := r.list(&podMonitorList); err != nil {
+		return fmt.Errorf("Could not list pod monitors: %v", err)
+	}
+	for i := range podMonitorList.Items {
+		obj := podMonitorList.Items[i]
+		utils.AddLabel(obj, monLabelKey, monLabelValue)
+		if err := r.update(obj); err != nil {
+			return err
+		}
+	}
+
+	serviceMonitorList := promv1.ServiceMonitorList{}
+	if err := r.list(&serviceMonitorList); err != nil {
+		return fmt.Errorf("Could not list service monitors: %v", err)
+	}
+	for i := range serviceMonitorList.Items {
+		obj := serviceMonitorList.Items[i]
+		utils.AddLabel(obj, monLabelKey, monLabelValue)
+		if err := r.update(obj); err != nil {
+			return err
+		}
+	}
+
+	promRuleList := promv1.PrometheusRuleList{}
+	if err := r.list(&promRuleList); err != nil {
+		return fmt.Errorf("Could not list prometheus rules: %v", err)
+	}
+	for i := range promRuleList.Items {
+		obj := promRuleList.Items[i]
+		utils.AddLabel(obj, monLabelKey, monLabelValue)
+		if err := r.update(obj); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+func (r *ManagedFusionReconciler) reconcilePrometheusProxyNetworkPolicy() error {
+	r.Log.Info("reconciling PrometheusProxyNetworkPolicy resources")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.prometheusProxyNetworkPolicy, func() error {
+		if err := r.own(r.prometheusProxyNetworkPolicy); err != nil {
+			return err
+		}
+		desired := templates.PrometheusProxyNetworkPolicyTemplate.DeepCopy()
+		r.prometheusProxyNetworkPolicy.Spec = desired.Spec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Failed to update prometheus proxy NetworkPolicy: %v", err)
+	}
+	return nil
+}
+
+func (r *ManagedFusionReconciler) initiateAgentUninstallation() error {
+	r.Log.Info("deleting agent namespace")
+	if err := r.delete(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: r.Namespace,
+		},
+	}); err != nil {
+		return fmt.Errorf("failed to delete namespace: %v", err)
+	}
+	r.Log.Info("Agent namespace deleted successfully")
```

Comment:
This message is incorrect, the fact that k8s API logged the request does not mean that the namespace is gone, Please remove it.

Reply:
Updated in latest commit